frontend/src/models/dashboardItemsModel.tsx

```diff
@@ -3,23 +3,34 @@ import { kea } from 'kea'
 import api from 'lib/api'
 import { prompt } from 'lib/logic/prompt'
 import { toast } from 'react-toastify'
-import { DashboardItemType } from '~/types'
+import { DashboardItemMode, DashboardItemType } from '~/types'
 import { dashboardsModel } from './dashboardsModel'
 import { Link } from 'lib/components/Link'
 import { dashboardItemsModelType } from '~/models/dashboardItemsModelType'
+import { insightLogic } from 'scenes/insights/insightLogic'
 
 export const dashboardItemsModel = kea<dashboardItemsModelType<DashboardItemType>>({
     actions: () => ({
         renameDashboardItem: (item: DashboardItemType) => ({ item }),
         renameDashboardItemSuccess: (item: DashboardItemType) => ({ item }),
+        updateDashboardItem: (id: number, payload: any) => ({ id, ...payload }),
         duplicateDashboardItem: (item: DashboardItemType, dashboardId?: number, move: boolean = false) => ({
             item,
             dashboardId,
             move,
         }),
+        setDashboardItemMode: (mode: DashboardItemMode) => ({ mode }),
         duplicateDashboardItemSuccess: (item: DashboardItemType) => ({ item }),
         refreshAllDashboardItems: (filters: Record<string, any>) => filters,
     }),
+    reducers: () => ({
+        dashboardItemMode: [
+            null,
```

Comment:
You probably want to set an explicit type here: `null as DashboardItemMode | null`. In general we should avoid implicit any's in typescript.